agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4422,29 +4422,37 @@ beerocks::message::sWifiChannel slave_thread::channel_selection_select_channel()
 
 bool slave_thread::channel_selection_current_channel_restricted()
 {
-    for (auto preference : channel_preferences) {
+    auto channel         = hostap_cs_params.channel;
+    auto bw              = static_cast<beerocks::eWiFiBandwidth>(hostap_cs_params.bandwidth);
+    auto operating_class = wireless_utils::get_operating_class_by_channel(channel, bw);
+
+    for (const auto &preference : channel_preferences) {
+        // for now we handle only non-operable preference
+        // TODO - handle as part of https://github.com/prplfoundation/prplMesh/issues/725
+        if (preference.preference.preference != 0) {
```

Comment:
Surely there must be a better name for this: `preference.preference.preference`
(nitpick)

Reply:
It bothers me too.

Reply:
Not part of this PR, but I can fix it in a separate commit. Will do.